nginx-sidecar/helm/nginx-sidecar/templates/_deployment-container.yaml

```
@@ -0,0 +1,54 @@
+{{- /*
+These template are meant to be injected inside a deployment object, as an additional container
+*/}}
+
+
+{{- /*
+This template is meant to be injected inside a deployment object.
+It needs to be passed an object that matches the structure of the values of this chart
+Example: include "nginx-sidecar.deployment-container.tpl" (index .Values "nginx-sidecar")
+*/}}
+{{- define "nginx-sidecar.deployment-container.tpl" }}
+- name: "{{ .objectNamePrefix }}"
+  image: "{{ .image.repository }}:{{ .image.tag }}"
```

can we default to the `.Chart.version`, and keep the values `image.tag` empty? Given how this function is used I'm not sure if this would work, or if `.Chart.version` would end up being the parent chart's version.
`image.pullPolicy` should be `IfNotPresent` by default in that case.

```suggestion
  image: "{{ .image.repository }}:{{ default .Chart.Version .image.tag }}"
```

---

Does this mean we need to tag automatically the image with the chart version when we build it?

---

Yes, the `cas-airflow-dag-trigger` does that, see https://github.com/bcgov/cas-airflow/blob/develop/.github/workflows/build-on-release.yaml

---

The problem here is that it's inside a template...so `.Chart.Version` won't resolve when the chart is imported. We'll need to put that in the values